.github/workflows/test.yaml

```diff
@@ -410,9 +414,8 @@ jobs:
         uses: actions/upload-artifact@v4
         if: always() && github.event_name == 'pull_request' && !cancelled()
         with:
-          name: report
+          name: report-${{ env.CONCURRENCY_GROUP}}
           path: report.json
-          overwrite: true
```

Comment:
Not sure if this is something you'd actually like -- but with calling this thing in a matrix style, each report kinda needs its own name.  The "overwrite:true" is a little flaky on github's end